addon/authenticators/oauth2-password-grant.js

```diff
@@ -346,10 +345,16 @@ export default BaseAuthenticator.extend({
             }));
           }
         });
+
+        const rejectWithResponse = this.get('rejectWithResponse');
+        deprecateNotRejectingWithResponse(rejectWithResponse);
```

Comment:
I'm not sure it makes sense to add this deprecation warning to places where it wasn't before, since we are not actually returning the response. Both here in `invalidate` and in `_refreshAccessToken` we are not returning the response (`_refreshAccessToken` is "private" even), there's nothing the user can do about it.

Reply:
I think you're confusing this with the new deprecation added in this PR. This is the deprecation we already had which is triggered when the `rejectWithResponse` property is `false`.

Reply:
Yes, but we are not really using the response in these two places (or sending it to the user)

Reply:
Yes, but the setting should **always** be `true` - if we're only checking for it in the `authenticate` method and trigger a deprecation there but someone overrides that method for example, they would maybe never see the deprecation although they rely on the deprecated behavior.

Reply:
Ok, I'm not entirely sure about showing this warning so much 😄  but I see your point.